### PR TITLE
Add ability to set member group, select member by uid and get all exi…

### PR DESCRIPTION
**Pull request by mnaichuk**
…sting groups

### CODE REVIEW

**Comment by pkucherenk0** [comment]
1. We probably should add search by group.
2. We probably shouldn't allow space or empty group (`" "`, `""`). We might dont want all symbols to be allowed. 
3. We might want to expose group to a user private api (user will be able to identify fees level) public api (users will be able to know the level system). Off course if there is no applogic to do it but i think peatio should expose thous.

**Comment by martianatwork** [comment]
I have a suggestion, 
We should be able to track back the fees group of the user, we can define fees group id in orders table.

File: app/models/member.rb [hunk 1]
```diff
@@ -27,6 +27,12 @@ class Member < ApplicationRecord
 
   attr_readonly :email
 
+  class << self
+    def groups
+      Member.pluck(:group).uniq
```
Comment (ysv) [inline app/models/member.rb]: I disagree with this 
1. To fetch all available member groups you should use `FeeGroup.pluck(:group)`

2. What is more I think right solution would be to select uniq on MySQL side
May be `Model.pluck("DISTINCT rating")` or `Model.distinct.pluck(:rating)`

3. Don't forget to append 'any'

**Review by ysv — CHANGES_REQUESTED** [review]

**Review by ysv — APPROVED** [review]

**Comment by ysv** [comment]
@martianatwork I would agree but what is the purpose of it ?
Since group, maker & taker may change it does not make sense to store reference to group that is why we store only fee percentage
It's still not ideal but we store  the most important data

**Comment by martianatwork** [comment]
@ysv 
currently we have to perform some logic to get the member's old group which slows down overall process.

It will be great to have an analytical data and easier way for the marketing team to see the performance of the marketing strategy or a fees group. 
this is based on our experience but i think other people will require the same as they grow.

**Pull request merged**